meta-lmp-base/recipes-support/resize-helper/resize-helper/resize-helper

```diff
@@ -23,40 +23,48 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
+# must exit on error
+set -e
+set -o pipefail
+
 # we must be root
 [ $(whoami) = "root" ] || { echo "E: You must be root" && exit 1; }
 
 # we must have few tools
 SGDISK=$(which sgdisk) || { echo "E: You must have sgdisk" && exit 1; }
-PARTED=$(which parted) || { echo "E: You must have parted" && exit 1; }
-PARTPROBE=$(which partprobe) || { echo "E: You must have partprobe" && exit 1; }
+FDISK=$(which fdisk) || { echo "E: You must have fdisk" && exit 1; }
+RESIZEPART=$(which resizepart) || { echo "E: You must have resizepart" && exit 1; }
 RESIZE2FS=$(which resize2fs) || { echo "E: You must have resize2fs" && exit 1; }
 
 # find root device
 ROOT_DEVICE=$(findmnt --noheadings --output=SOURCE / | cut -d'[' -f1)
 # prune root device (for example UUID)
 ROOT_DEVICE=$(realpath ${ROOT_DEVICE})
 # get the partition number and type
-PART_ENTRY_NUMBER=$(udevadm info --query=property --name=${ROOT_DEVICE} | grep '^ID_PART_ENTRY_NUMBER=' | cut -d'=' -f2)
+INFO=$(udevadm info --query=property --name=${ROOT_DEVICE})
+PART_ENTRY_NUMBER=$(echo "${INFO}" | grep '^ID_PART_ENTRY_NUMBER=' | cut -d'=' -f2)
 
 # in case the root device is not on a partitioned media
 if [ "x$PART_ENTRY_NUMBER" = "x" ]; then
 	${RESIZE2FS} "${ROOT_DEVICE}"
 	exit 0
 fi
 
-PART_TABLE_TYPE=$(udevadm info --query=property --name=${ROOT_DEVICE} | grep '^ID_PART_TABLE_TYPE=' | cut -d'=' -f2)
+PART_TABLE_TYPE=$(echo "${INFO}" | grep '^ID_PART_TABLE_TYPE=' | cut -d'=' -f2)
 # find the block device
 DEVICE=$(udevadm info --query=path --name=${ROOT_DEVICE} | awk -F'/' '{print $(NF-1)}')
 DEVICE="/dev/${DEVICE}"
 
+SIZE=$(blockdev --getsz ${DEVICE})
 if [ "$PART_TABLE_TYPE" = "gpt" ]; then
 	${SGDISK} -e ${DEVICE}
-	${PARTPROBE}
+	SIZE=$(($SIZE - 33)) # the GPT end of disk is 34 sectors
 fi
 
-# 'Yes' is required when partition is already mounted
-echo -e 'Yes\n100%' | ${PARTED} ---pretend-input-tty ${DEVICE} resizepart ${PART_ENTRY_NUMBER} ||
-	echo -e '100%' | ${PARTED} ---pretend-input-tty ${DEVICE} resizepart ${PART_ENTRY_NUMBER}
-${PARTPROBE}
+# Use fdisk to repartition
+END=$((${SIZE} - 1))
+PARTOF=$(echo "${INFO}" | grep '^ID_PART_ENTRY_OFFSET=' | cut -d'=' -f2)
+echo -e "d\n${PART_ENTRY_NUMBER}\nn\np\n${PART_ENTRY_NUMBER}\n${PARTOF}\n${END}\nw\n" | ${FDISK} ${DEVICE} > /dev/null
```

Comment:
Is it so verbose that it is better to dump to /dev/null? I would prefer for it to go to stdout as we can then see what happened via journal.

Reply:
Please add a comment as well saying what it is being done here (deleting partition and creating a new one up to END)

Reply:
And we do need partprobe after this gets called.

Reply:
Another question, END here is in 512-byte sectors while we might have disks with larger sectors (e.g. 4096), is fdisk assuming 512-byte sectors by default?